src/nvim/move.c

```diff
@@ -146,9 +146,14 @@ void update_topline(void)
   bool check_botline = false;
   long save_so = p_so;
 
+  win_grid_alloc(curwin, false);
+  if (!screen_valid(true)) {
+    return;
+  }
+
   // If there is no valid screen and when the window height is zero just use
   // the cursor line.
-  if (!screen_valid(true) || curwin->w_height == 0) {
+  if (curwin->w_grid.Rows == 0) {
```

Comment:
@UtkarshMe The patch that added the now failing `test_startup.vim` test added `screen_valid` here, why was it removed again?

Reply:
I remember changing it because some test was failing because of it. But now, reverting and re-running the tests, all seem to pass. Maybe there was some other problem which was causing it. Reverting back!